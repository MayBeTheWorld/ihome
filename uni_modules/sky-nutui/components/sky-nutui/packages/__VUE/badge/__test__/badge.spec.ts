import Badge from '../index.vue';
import { mount } from '@vue/test-utils';

test('should render nothing when content is empty string', () => {
  const wrapper = mount(Badge, {
    props: {
      value: ''
    }
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should render nothing when content is undefined', () => {
  const wrapper = mount(Badge, {
    props: {
      value: undefined
    }
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should render nothing when content is zero', () => {
  const wrapper = mount(Badge, {
    props: {
      value: 0
    }
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should render content slot correctly', () => {
  const wrapper = mount(Badge, {
    slots: {
      default: () => 'Custom Content'
    }
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should change dot position when using offset prop', () => {
  const wrapper = mount(Badge, {
    props: {
      top: '4',
      right: '-2'
    }
  });

  const badge: any = wrapper.find('.nut-badge');
  expect(badge.element.style.top).toMatchSnapshot('4px');
  expect(badge.element.style.right).toMatchSnapshot('-2px');
});
