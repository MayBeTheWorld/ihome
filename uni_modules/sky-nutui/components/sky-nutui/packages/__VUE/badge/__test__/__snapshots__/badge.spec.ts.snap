// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should change dot position when using offset prop: -2px 1`] = `""`;

exports[`should change dot position when using offset prop: 4px 1`] = `""`;

exports[`should render content slot correctly 1`] = `
"<view class=\\"nut-badge\\">
  <view class=\\"slot-icons\\"></view>Custom Content<view class=\\"nut-badge__content sup\\" style=\\"top: 0px; right: 0px; z-index: 9; display: none;\\"></view>
</view>"
`;

exports[`should render nothing when content is empty string 1`] = `
"<view class=\\"nut-badge\\">
  <view class=\\"slot-icons\\"></view>
  <view class=\\"nut-badge__content sup\\" style=\\"top: 0px; right: 0px; z-index: 9; display: none;\\"></view>
</view>"
`;

exports[`should render nothing when content is undefined 1`] = `
"<view class=\\"nut-badge\\">
  <view class=\\"slot-icons\\"></view>
  <view class=\\"nut-badge__content sup\\" style=\\"top: 0px; right: 0px; z-index: 9; display: none;\\"></view>
</view>"
`;

exports[`should render nothing when content is zero 1`] = `
"<view class=\\"nut-badge\\">
  <view class=\\"slot-icons\\"></view>
  <view class=\\"nut-badge__content sup\\" style=\\"top: 0px; right: 0px; z-index: 9; display: none;\\">0</view>
</view>"
`;
